docs/book/src/developer/repository-layout.md

```diff
@@ -1,3 +1,175 @@
+
+### Cluster API Provider Docker 
+
+[~/test/insfrastructure/docker](https://github.com/kubernetes-sigs/cluster-api/tree/main/test/infrastructure/docker)
+
+This folder contains a kubebuilder project i.e. CAPD which is a reference implementation of an infrastructure provider for the Cluster API project using Docker. This provider is intended for development purposes only. 
+
+Cluster API has all the three components needed to run a Cluster API management cluster - 
+CABPK (bootstrap provider), KCP (KubeAdm Control Plane provider) and CAPD (Infrastructure provider using Docker). Hence, Cluster API project has all batteries included. All these components are released together along with CAPI releases. This simplifies the release process significantly as we do not have to manage and coordinate releases for each of these providers separately.
+
+### Clusterctl CLI
+
+[~/cmd/clusterctl](https://github.com/kubernetes-sigs/cluster-api/tree/main/cmd/clusterctl)
+
+This folder contains Clusterctl, a CLI which is used to simplify the bootstrapping of a  Cluster API management cluster by providing easy interface to the users.
+
+### Configuration
+
+[~/config](https://github.com/kubernetes-sigs/cluster-api/tree/main/config)
+
+This is a Kubernetes manifest folder containing application resource configuration as kustomize YAML definitions. Some of the subfolders are:
+* [~/config/certmanager](https://github.com/kubernetes-sigs/cluster-api/tree/main/config/certmanager) - It contains manifests like self-signed issuer CR and certificate CR useful for cert manager.
+
+* [~/config/crd](https://github.com/kubernetes-sigs/cluster-api/tree/main/config/crd) - It contains CRDs generated from types defined in [api](#api) folder
+	
+* [~/config/manager](https://github.com/kubernetes-sigs/cluster-api/tree/main/config/manager) - It contains manifest for the deployment of core Cluster API manager.
+	
+* [~/config/rbac](https://github.com/kubernetes-sigs/cluster-api/tree/main/config/rbac) - Manifests for RBAC resources generated from kubebuilder markers defined in controllers.
+	
+* [~/config/webhook](https://github.com/kubernetes-sigs/cluster-api/tree/main/config/webhook) - Manifest for webhooks generated from the markers defined in the web hook implementations present in [api](#api) folder.
+
+### Controllers
+
+[~/internal](https://github.com/kubernetes-sigs/cluster-api/tree/main/internal)
+
+This folder contains resources which are not meant to be used directly by users of Cluster API. For e.g. the implementation of controllers is present in [~/internal/controllers](https://github.com/kubernetes-sigs/cluster-api/tree/main/internal/controllers) directory so that we can make changes in controller implementation without breaking users. This allows us to keep our api surface smaller and move faster.
```

Comment:
```suggestion
This folder contains resources which are not meant to be used directly by users of Cluster API. e.g. the implementation of controllers is present in [~/internal/controllers](https://github.com/kubernetes-sigs/cluster-api/tree/main/internal/controllers) directory so that we can make changes in controller implementation without breaking users. This allows us to keep our api surface smaller and move faster.
```